### PR TITLE
ErrorHandler removed

### PR DESCRIPTION
**Pull request by rodrigofeijao**
ErrorHandler is now optional, but set to be used by default.

The `swagger()` now accepts `Options`, an object wich can include an array of middlewares.

### CODE REVIEW

File: src/swagger.ts [hunk 1]
```diff
@@ -82,18 +85,32 @@ function defineRoutes(paths: OpenAPIV2.PathsObject, controllers: Controllers): R
   return router;
 }
 
-export default function swagger(swaggerFile: string, controllers: Controllers): Router {
+const defaultOptions: Options = {
+  middleware: [],
```
Comment (igoratron) [inline src/swagger.ts:89]: preMiddleware? because it's run before the main swagger middleware?

File: src/swagger.ts [hunk 2]
```diff
@@ -82,18 +85,32 @@ function defineRoutes(paths: OpenAPIV2.PathsObject, controllers: Controllers): R
...
+  useInternalErrorHandler: true,
```
Comment (igoratron) [inline src/swagger.ts:90]: do we even need it here? maybe we should remove it altogether?

File: src/swagger.ts (hunk 1 above)
Comment (rodrigofeijao) [reply src/swagger.ts:89]: good point!

File: src/swagger.ts (hunk 2 above)
Comment (rodrigofeijao) [reply src/swagger.ts:90]: I was aiming to stage it. Not sure how PG is using it, so in order to don't break everything on their end, I was trying to make it optional first, and then if they don't use it as well, we remove it.

Comment (rodrigofeijao) [reply src/swagger.ts]: We've decided to remove this.

**Pull request merged**